Issue building with JupyterLab 2.1.1 and tsc 3.9.3

When trying to build master with JupyterLab 2.1.1 and tsc 3.9.3, the following error happens:

```
[4/4] Building fresh packages...
$ tsc --build packages/*
packages/dataregistry/src/urltemplates.ts:91:7 - error TS2322: Type '{ [x: string]: NonNullable<T[string]>; }' is not assignable to type 'T'.
  '{ [x: string]: NonNullable<T[string]>; }' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint '{ [arg: string]: any; }'.

91       return newObject;
         ~~~~~~~~~~~~~~~


Found 1 error.

error Command failed with exit code 1.
```

It can be fixed by casting newObject to T.

--- a/packages/dataregistry/src/urltemplates.ts
+++ b/packages/dataregistry/src/urltemplates.ts
@@ -85,7 +85,7 @@ export class URLTemplate<T extends { [arg: string]: any }> extends TypedURL<T> {
     }
     const newObject = Object.fromEntries(
       Object.entries(args).map(([key, val]) => [key, this.map[key][0](val)])
-    );
+    ) as T;
     // If none  of the value were none, return the object
     if (nonNullableValues(newObject)) {
       return newObject;

